fs_mol/utils/protonet_utils.py

```diff
@@ -200,46 +270,22 @@ def train_loop(self, out_dir: str, dataset: FSMolDataset, aml_run=None):
             )
 
             if step % self.config.validate_every_num_steps == 0:
-                valid_task_sample_iterable = get_protonet_task_sample_iterable(
-                    dataset=dataset,
-                    data_fold=DataFold.VALIDATION,
-                    num_samples=1,
-                    max_num_graphs=self.config.batch_size,
-                    support_size=self.config.support_set_size,
-                    query_size=self.config.query_set_size,
-                )
 
-                valid_losses: List[float] = []
-                valid_metrics: List[BinaryEvalMetrics] = []
-                for task_sample in valid_task_sample_iterable:
-                    task_loss, task_metrics = run_on_batches(
-                        self,
-                        task_sample.batches,
-                        batch_labels=task_sample.batch_labels,
-                        train=False,
-                    )
-                    valid_losses.append(task_loss)
-                    valid_metrics.append(task_metrics)
-                mean_valid_loss = np.mean(valid_losses)
-                avg_valid_metrics = avg_metrics_list(valid_metrics)
-
-                valid_avg_prec_mean, valid_avg_prec_std = avg_valid_metrics["avg_precision"]
+                valid_metric = validate_by_finetuning_on_tasks(self, dataset)
 
                 if aml_run:
                     # printing some measure of loss on all validation tasks.
-                    aml_run.log(f"valid_mean_loss", mean_valid_loss)
-                    aml_run.log(f"valid_mean_avg_prec", valid_avg_prec_mean)
-                    aml_run.log(f"valid_mean_kappa", avg_valid_metrics["kappa"][0])
+                    aml_run.log(f"valid_mean_avg_prec", valid_metric)
```

Comment:
Can we keep logging all of these values by returning all validation metrics and unpacking the necessary bit on the outside?

Reply:
I think these are actually all aml_run logged in the new validate_by_finetuning_on_tasks function.